pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/MethodReturnsInternalArray.xml

```diff
@@ -266,4 +266,116 @@ public enum MethodReturnsInternalArrayCaseEnum {
         ]]></code>
     </test-code>
 
+    <test-code>
+        <description>Inner interface</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public interface OuterInterface {
+       interface InnerInterface {
+            String method();
+       }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Inner annotation</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Retention;
+
+public final class I {
+    private I(){
+    }
+    @Retention(RetentionPolicy.RUNTIME)
+    public static @interface Inner {}
```

Comment:
Same here, we'll need something with an array:

```java
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Retention;
public final class I {
    private I(){
    }
    @Retention(RetentionPolicy.RUNTIME)
    public static @interface Inner {
        String[] value();
    }
}
```

Reply:
done